cli/src/cli.rs

```diff
@@ -0,0 +1,100 @@
+use {
+    crate::{command::Command, helper::CliHelper, print::Print},
+    gluesql::{
+        prelude::Glue,
+        store::{GStore, GStoreMut},
+    },
+    rustyline::{error::ReadlineError, Editor},
+    std::{
+        fmt::Debug,
+        io::{Result, Write},
+        marker::PhantomData,
+    },
+};
+
+pub struct Cli<T, U, W>
+where
+    T: Debug,
+    U: GStore<T> + GStoreMut<T>,
+    W: Write,
+{
+    glue: Glue<T, U>,
+    print: Print<W>,
+    _marker: PhantomData<T>,
+}
+
+impl<T, U, W> Cli<T, U, W>
+where
+    T: Debug,
+    U: GStore<T> + GStoreMut<T>,
+    W: Write,
+{
+    pub fn new(storage: U, output: W) -> Self {
+        let glue = Glue::new(storage);
+        let print = Print::new(output);
+
+        Self {
+            glue,
+            print,
+            _marker: PhantomData,
+        }
+    }
+
+    pub fn run(&mut self) -> Result<()> {
+        macro_rules! println {
+            ($($p:tt),*) => ( writeln!(&mut self.print.output, $($p),*)?; )
+        }
```

Comment:
[Question]
Is there a reason to redefine println macro ?

Reply:
Perhaps this is to reuse `writeln!(&mut ...)`.

Reply:
The only reason is simplifying `&mut self.print.output`.
Currently, `cli.rs` does not have unit test but this can be used for it.

Reply:
Aha!